packages/logs/lib/models/messages.ts

```diff
@@ -249,20 +253,46 @@ export async function listMessages(opts: {
         });
     }
 
+    // Sort and cursor
+    let cursor: any[] | undefined;
+    let sort: estypes.Sort = [{ createdAt: 'desc' }, { id: 'desc' }];
+    if (opts.cursorBefore) {
+        cursor = parseCursor(opts.cursorBefore);
+        sort = [{ createdAt: 'asc' }, { id: 'asc' }];
+    } else if (opts.cursorAfter) {
+        cursor = opts.cursorAfter ? parseCursor(opts.cursorAfter) : undefined;
+    }
+
     const res = await client.search<MessageRow>({
         index: indexMessages.index,
         size: opts.limit,
-        sort: [{ createdAt: 'desc' }, 'id'],
+        sort,
         track_total_hits: true,
+        search_after: cursor,
         query
     });
     const hits = res.hits;
 
+    const total = typeof hits.total === 'object' ? hits.total.value : hits.hits.length;
+    const totalPage = hits.hits.length;
+    const items = hits.hits.map((hit) => {
+        return hit._source!;
+    });
+    if (opts.cursorBefore) {
+        // In case we set before we have to reverse the message since we inverted the sort
```

Comment:
Can we add a link to https://github.com/elastic/elasticsearch/issues/29449 Might be handy if we ever run into the limitation mentioned in the issue